routers/web/org/teams.go

```diff
@@ -124,10 +125,18 @@ func TeamsAction(ctx *context.Context) {
 				return
 			}
 		}
-		ctx.JSON(http.StatusOK,
-			map[string]any{
-				"redirect": ctx.Org.OrgLink + "/teams/" + url.PathEscape(ctx.Org.Team.LowerName),
-			})
+		redirect := ctx.Org.OrgLink + "/teams/" + url.PathEscape(ctx.Org.Team.LowerName)
+		if isOrgMember, err := org_model.IsOrganizationMember(ctx, ctx.Org.Organization.ID, ctx.Doer.ID); err != nil {
+			ctx.ServerError("IsOrganizationMember", err)
+			return
+		} else if !isOrgMember {
+			if ctx.Org.Organization.Visibility.IsPrivate() {
+				redirect = setting.AppSubURL + "/"
+			} else {
+				redirect = ctx.Org.Organization.HomeLink()
+			}
+		}
+		ctx.JSONRedirect(redirect)
```

Comment:
What about extracting this check into a separate method and giving it a parameter `defaultRedirect`?
You call it twice at the moment.

Reply:
Done.